threescaleAPI-operator/pkg/apis/3scale/v1alpha1/types.go

```diff
@@ -24,9 +24,10 @@ type API struct {
 }
 
 type The3ScaleConfig struct {
-	AccessToken       string `json:"AccessToken"`
-	AdminPortalURL    string `json:"AdminPortalURL"`
-	IntegrationMethod string `json:"IntegrationMethod"`
+	Secret            *CredentialSecret `json:"secret,omitempty"`
```

Comment:
What about .. 


With accessToken / SystemURL: 

```
3scaleConfig:
    IntegrationMethod: apicast
    Credentials:    
       - systemURL: https://aaaaa-admin.3scale.net
          accessToken: XXXXXXXXXXXX     
Upstream: https://myUnprotectedAPI:80
OpenAPIDefinition: aaaaaaa
```

With Secret
```
3scaleConfig:
    IntegrationMethod: apicast
    Credentials:    
      - secret:
            name: aaaaaaaaaaa
Upstream: https://myUnprotectedAPI:80
OpenAPIDefinition: aaaaaaa
```

wdyt?

Reply:
Yeah, I like that actually its less ambiguous. Refactoring ...